content/docs/reference/pkg/aws/s3/bucket.md

```diff
@@ -0,0 +1,10931 @@
+A rule of [Cross-Origin Resource Sharing](https://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html) (documented below).
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">force_<wbr>destroy</td>
+            <td class="align-top">
+                
+                <code>bool</code>
+            </td>
+            <td class="align-top">{{% md %}} 
+ (Optional)
+A boolean that indicates all objects (including any [locked objects](https://docs.aws.amazon.com/AmazonS3/latest/dev/object-lock-overview.html)) should be deleted from the bucket so that the bucket can be destroyed without error. These objects are *not* recoverable.
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">grants</td>
+            <td class="align-top">
+                
+                <code><a href="#bucketgrant">list[bucket_<wbr>grant]</a></code>
+            </td>
+            <td class="align-top">{{% md %}} 
+ (Optional)
+An [ACL policy grant](https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#sample-acl) (documented below). Conflicts with `acl`.
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">hosted_<wbr>zone_<wbr>id</td>
+            <td class="align-top">
+                
+                <code>str</code>
+            </td>
+            <td class="align-top">{{% md %}} 
+ (Optional)
+The [Route 53 Hosted Zone ID](https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints) for this bucket&#39;s region.
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">lifecycle_<wbr>rules</td>
+            <td class="align-top">
+                
+                <code><a href="#bucketlifecyclerule">list[bucket_<wbr>lifecycle_<wbr>rule]</a></code>
+            </td>
+            <td class="align-top">{{% md %}} 
+ (Optional)
+A configuration of [object lifecycle management](http://docs.aws.amazon.com/AmazonS3/latest/dev/object-lifecycle-mgmt.html) (documented below).
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">loggings</td>
+            <td class="align-top">
+                
+                <code><a href="#bucketlogging">list[bucket_<wbr>logging]</a></code>
+            </td>
+            <td class="align-top">{{% md %}} 
+ (Optional)
+A settings of [bucket logging](https://docs.aws.amazon.com/AmazonS3/latest/UG/ManagingBucketLogging.html) (documented below).
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">object_<wbr>lock_<wbr>configuration</td>
+            <td class="align-top">
+                
+                <code><a href="#bucketobjectlockconfiguration">dict{bucket_<wbr>object_<wbr>lock_<wbr>configuration}</a></code>
```

Comment:
This syntax for dictionaries doesn't feel very Pythony, but I don't have a great alternative to suggest. I guess technically: `Dict[str, Any]`.

If we bumped our min version of Python from 3.6 to 3.8, we could consider introducing `TypedDict` for these, which means, the nested type could be emitted for Python as something like:

```python
class BucketObjectLockConfiguration(TypedDict):
    objectLockEnabled: str
    rule: Optional[BucketObjectLockConfigurationRule]
```

or

```python
class BucketObjectLockConfiguration(TypedDict, total=False):
    objectLockEnabled: str
    rule: BucketObjectLockConfigurationRule
```

And then we could just refer to it here as `BucketObjectLockConfiguration`.

Reply:
Agreed - we should use `PascalCase` names for Python types.